### PR TITLE
Add ExternalName to service docs

### PR DESCRIPTION
**Pull request by therc**
This is the doc part of  https://github.com/kubernetes/features/issues/33

cc @kubernetes/docs @thockin @smarterclayton

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/kubernetes.github.io/1063)

### CODE REVIEW

File: docs/user-guide/services/index.md [hunk 1]
```diff
@@ -147,11 +147,38 @@ Accessing a `Service` without a selector works the same as if it had selector.
...
+{
+    "kind": "Service",
+    "apiVersion": "v1",
+    "metadata": {
+        "name": "my-service",
+        "namespace": "prod"
+    },
+    "spec": {
+        "externalName": "my.database.example.com",
+        "type": "ExternalName"
```
Comment (thockin) [inline docs/user-guide/services/index.md]: put type first

**Comment by thockin** [comment]
Let's queue this up until https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30599 and https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30931 are in

**Comment by thockin** [comment]
needs a rebase, sorry

**Comment by jaredbhatti** [comment]
Pingity ping.  Can you rebase? I'll merge your request afterwards.

**Comment by therc** [comment]
Squashed and rebased

**Comment by thockin** [comment]
@jaredbhatti how are we doing doc merges?  Me?  You?  Automatic?

**Comment by thockin** [comment]
@jaredbhatti @devin-donnelly @kubernetes/docs

**Comment by thockin** [comment]
need rebase again, sorry

**Comment by therc** [comment]
Rebased.

**Comment by thockin** [comment]
git disagrees...

**Comment by therc** [comment]
Why does it need a rebase? I don't see any "needs rebase" label here as in the main repo (unfortunate, but helpful). I just did another pull/rebase/push.

**Comment by thockin** [comment]
github said it was out of sync and needed a rebase... it doesn't tell me why..

**Comment by thockin** [comment]
@kubernetes/docs ready for doc review

**Comment by therc** [comment]
Can this be merged? Anything delaying it still? I am travelling and will have poor connectivity for the rest of the week.

File: docs/user-guide/services/index.md [hunk 2]
```diff
@@ -147,11 +147,38 @@ Accessing a `Service` without a selector works the same as if it had selector.
...
+the redirection happens at the DNS level and no proxying or forwarding occurs.
+Should you later decide to move your database into your cluster, you can start
+its pods, add appropriate selectors or endpoints and change the service `type`.
+
 ## Virtual IPs and service proxies
 
-Every node in a Kubernetes cluster runs a `kube-proxy`.  This application
-is responsible for implementing a form of virtual IP for `Service`s.  In
-Kubernetes v1.0 the proxy was purely in userspace.  In Kubernetes v1.1 an
+Every node in a Kubernetes cluster runs a `kube-proxy`.  This application is
```
Comment (devin-donnelly) [inline docs/user-guide/services/index.md]: Pronoun avoidance.

"Every node in a Kubernetes cluster runs a `kube-proxy`. `kube-proxy` is responsible for..."

File: docs/user-guide/services/index.md [hunk 3]
```diff
@@ -343,7 +373,10 @@ records (addresses) which point directly to the `Pods` backing the `Service`.
 
 For headless services that do not define selectors, the endpoints controller does
 not create `Endpoints` records. However, the DNS system looks for and configures
-A records for any `Endpoints` that share a name with the service.
+either
```
Comment (devin-donnelly) [inline docs/user-guide/services/index.md]: Add a colon (:) after "either".

File: docs/user-guide/services/index.md [hunk 4]
```diff
@@ -147,11 +147,38 @@ Accessing a `Service` without a selector works the same as if it had selector.
 The traffic will be routed to endpoints defined by the user (`1.2.3.4:9376` in
 this example).
 
+A special case of service without selectors is an ExternalName service.
```
Comment (devin-donnelly) [inline docs/user-guide/services/index.md]: "An ExternalName service is a special case of service that does not have selectors."

**Comment by devin-donnelly** [comment]
A few comments on Docs Review. Also, since this PR should not be merged and published to master until 1.4 is released, can you change the PR to target the release-1.4 branch?

**Comment by therc** [comment]
Incorporated fixes, changed to target release-1.4.

---

Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[docs/user-guide/services/index.md, line 164 at r1](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1063#-KPZ69IqMf-hx8aCCzOu-r1-164:-KRbbGlNP5wzqARW2TJt:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/67a3a266f2cc399e8da056687bd49436ce5f4a53/docs/user-guide/services/index.md#L164)):*
<details><summary><i>Previously, thockin (Tim Hockin) wrote…</i></summary>
> put type first
</details>
Done.

---

*[docs/user-guide/services/index.md, line 150 at r2](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1063#-KPZ69IqMf-hx8aCCzOu-r2-150:-KRbbFlkol-jdxv_RJd1:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/1c898c6918764a7a4012a0dd99a93e3342f2a037/docs/user-guide/services/index.md#L150)):*
<details><summary><i>Previously, devin-donnelly wrote…</i></summary>
> "An ExternalName service is a special case of service that does not have selectors."
</details>
Done.

---

*[docs/user-guide/services/index.md, line 178 at r2](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1063#-KPZ69IqMf-hx8aCCzOu-r2-178:-KRbbI-fScaqBRMmQor6:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/1c898c6918764a7a4012a0dd99a93e3342f2a037/docs/user-guide/services/index.md#L178)):*
<details><summary><i>Previously, devin-donnelly wrote…</i></summary>
> Pronoun avoidance.
> 
> 
> 
> "Every node in a Kubernetes cluster runs a `kube-proxy`. `kube-proxy` is responsible for..."
</details>
Done.

---

*[docs/user-guide/services/index.md, line 376 at r2](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1063#-KPZ69IqMf-hx8aCCzOu-r2-376:-KRbbItBfRpAhdrFtGIV:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/1c898c6918764a7a4012a0dd99a93e3342f2a037/docs/user-guide/services/index.md#L376)):*
<details><summary><i>Previously, devin-donnelly wrote…</i></summary>
> Add a colon (:) after "either".
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1063#-:-KRbbMSADlGTMNe5jdpS:b-4hj6gd)*

**Pull request merged**